agent/tests/test_streamsets_with_preferred_type.py

```diff
@@ -0,0 +1,42 @@
+import time
+
+import pytest
+
+from agent import cli, streamsets, pipeline
+from agent.cli.run_test_pipeline import perform_cleanup
+from .conftest import generate_input
+
+url = 'http://dc2:18630'
+
+
+def test_streamsets_1(cli_runner):
+    args = ['--url', url, '--preferred-type', 'directory']
+    result = cli_runner.invoke(cli.streamsets.add, args=args, catch_exceptions=False)
+    streamsets.repository.get_by_url(url)
+    assert result.exit_code == 0
+
+
+def test_edit_streamsets(cli_runner):
+    input_ = {
+        'username': '',
+        'password': '',
+        'agent_external_url': '',
+    }
+    result = cli_runner.invoke(cli.streamsets.edit, [url], catch_exceptions=False, input=generate_input(input_))
+    streamsets.repository.get_by_url(url)
+    assert result.exit_code == 0
+
+
+def test_run_pipeline(cli_runner):
+    result = cli_runner.invoke(cli.run_test_pipeline, args=['--no-delete'], catch_exceptions=False)
+    print(f"run_test_pipeline: {result.exit_code} {result.output}")
+    assert pipeline.repository.get_by_streamsets_url(url)
```

Comment:
What we are asserting here? List not empty?

Reply:
At this case, db contain 2 streamsets, and we check that pipeline with directory type is assigned to streamsets with directory type

Reply:
I mean usually, with `assert` statements we evaluate some expression. Smth like:
```python
assert len(pipeline.repository.get_by_streamsets_url(url)) == 1
```

Reply:
Ok, I have idea how rework this test